Xamarin.Forms.Core/Application.cs

```diff
@@ -169,6 +169,10 @@ public ResourceDictionary Resources
 
 		public event EventHandler<ModalPushingEventArgs> ModalPushing;
 
+		public event EventHandler<Page> PageAppearing;
```

Comment:
it those events aren't meant to be consumed by apps, but by tools, make them internal, access then through reflection.

Reply:
agreed with @jassmith that it should be public since other customers have requested such a feature in the past.